src/omv/alloc/fb_alloc.c

```diff
@@ -132,7 +132,15 @@ void *fb_alloc(uint32_t size, int hints)
         return NULL;
     }
 
-    size=((size+sizeof(uint32_t)-1)/sizeof(uint32_t))*sizeof(uint32_t);// Round Up
+    size = ((size + sizeof(uint32_t) - 1) / sizeof(uint32_t)) * sizeof(uint32_t); // Round Up
+
+    #if __DCACHE_PRESENT
+    if (hints & FB_ALLOC_CACHE_ALIGN) {
+        size = ((size + __SCB_DCACHE_LINE_SIZE - 1) / __SCB_DCACHE_LINE_SIZE) * __SCB_DCACHE_LINE_SIZE;
+        size += __SCB_DCACHE_LINE_SIZE;
```

Comment:
```suggestion
```
Why do you add another 32 bytes it's already a multiple of 32. Any example why this is needed ?

Reply:
This is needed for the offset code below. Here's what's happening:

size = ((size + __SCB_DCACHE_LINE_SIZE - 1) / __SCB_DCACHE_LINE_SIZE) * __SCB_DCACHE_LINE_SIZE;

Ensures that you are using up some number of cache lines. This makes sure that no data at the end of the buffer will be inside another cache line. However, we don't know if the data will start on a cache line.

size += __SCB_DCACHE_LINE_SIZE;

Gives us extra memory that will be eaten by the offset code below. We have no idea what alignment of RAM fb_alloc() may give us. So, there's no way to know how much of __SCB_DCACHE_LINE_SIZE will be used up. We could be aligned perfectly by default, or, 4, 8, 12, 16, 20, 24, or 28 bytes offset. If we are offset by an number then the offset code below will push the pointer forward and eat 28, 24, 20, 16, 12, 8, or 4 bytes respectively. In the case where we are perfectly aligned we use 32 bytes extra.

Back to: size = ((size + __SCB_DCACHE_LINE_SIZE - 1) / __SCB_DCACHE_LINE_SIZE) * __SCB_DCACHE_LINE_SIZE;

Now that the pointer returned is aligned to a cache line... then because we did the above we are using N number of cache lines. While this means we will have some unused data at the end of the cache line it's okay as if the processor accesses memory around the DMA buffer it won't end up caching part of any of the DMA buffer causing problems.